api/src/services/taxonomy-service.ts

```diff
@@ -19,11 +19,20 @@ export class TaxonomyService {
 
   private sanitizeSpeciesData = (data: SearchHit<any>[]) => {
     return data.map((item) => {
-      const unit_name1 = item._source.unit_name1 || '';
-      const unit_name2 = item._source.unit_name2 || '';
-      const unit_name3 = item._source.unit_name3 || '';
-      const english_name = item._source.english_name || '';
-      const label = `${item._source.code}: ${item._source.tty_kingdom} ${item._source.tty_name}, ${unit_name1} ${unit_name2} ${unit_name3}, ${english_name}`;
+      const label = [
+        item._source.code,
+        [
+          [item._source.tty_kingdom || '', item._source.tty_name || ''].filter(Boolean).join(' '),
```

Comment:
If `item._source.tty_kingdom` is "falsy", why should we coalesce it to empty string, which is also falsy? Both will get dropped by the `Boolean` filter, but if `item._source.tty_kingdom` is truthy then it will never coalesce to empty string.